### ISSUE TITLE
Load test Linky

### ISSUE THREAD

**Issue by swirtSJW** [department-of-veterans-affairs/va.gov-cms#9677]
## Description

Adding Linky to turn all external links into entities, has a great potential for us to be able to audit and manipulate external links in the same way we can with internal links on using entity_useage module (not in the codebase yet).

However, turning all external links into entities is likely to create a performance hit as it increases the number of database queries it takes to render a page.  On the day to day interaction with the CMS, this is not likely to be an issue.

Where it may cause a noticeable issue is on the graphQL queries used to build the FE.   Will also likely impact jsonapi queries for Accelerated publishing.

Testing will need to:
1. Fully linkify all node content (VBO did not seem to address this.  May need a drush command) 
2. Time test the graphQL queries that build the FE and compare the time to non-linky environment.
3. Consult with AP team on how to test the jsonapi implementations they are working on.

We can likely use this PR environment https://github.com/department-of-veterans-affairs/va.gov-cms/pull/9575

## Acceptance Criteria
- [ ] Time testing FE graphQL queries to quantify the performance
- [ ] ?  Time testing jasonapi queries???
- [ ] Making a decision...linky or no linky
- [ ] Platform understands their role (new github issue?)

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [x] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by EWashb** [comment]
We need to get with Platform and AP to determine if this change affects them

**Comment by timcosgrove** [comment]
We would like to see the performance testing methods and results before Sitewide decides to install it. I'm also curious about the UX of this:

> That way if your link changes, you update it once and it is updated through all content.

That sounds nice but also possibly confusing. Curious how it handles multiple entities with the same link (it might disallow?), links with similar structures but different query strings, etc.

**Comment by EWashb** [comment]
@tjheffner could you sync with @timcosgrove on the items above? Also happy to help as much as I can

**Comment by tjheffner** [comment]
Taking a look at this today:

> Fully linkify all node content (VBO did not seem to address this. May need a drush command)

I'm not seeing a drush command available to do this. I do see this module that adds to Linky: https://www.drupal.org/project/lms but it doesn't look like there's any active work being done at this time. Maybe an opportunity for us to push back to open source.

> We would like to see the performance testing methods and results before Sitewide decides to install it.

Yes, Sitewide will not be installing the module until performance testing is concluded and deemed satisfactory by all affected parties (Sitewide, FE graphql consumers, AP) This is basically a large sub-task of the #9424 SPIKE.

Planned testing: 
- measure the time of the graphql portion when building the FE currently from up-to-date main branches.
- after all links have been linkified (either in the linked PR environment above (#9575) or somewhere else), measure the time of the graphql portion when building the FE with these new link entities.
- repeat previous steps for json:api
- compare query times of current cms content times vs proposed-linkified content times

@timcosgrove I'll find a short slot on your calendar for when you're feeling better and we can talk it through further if needed :)

**Comment by EWashb** [comment]
I think we have a path forward here @tjheffner. Do we need to make another ticket for the next step so that this one can be closed?

**Comment by EWashb** [comment]
current build in a meaningful way and therefore we wouldn't get useful information regarding performance if we load test it before an initial migration of external links is done. We need to actually get a sizeable number of links managed with linky in order to determine. The next step is to write a script to convert all existing external links. Once that is done, we will have the data to load test and move forward or rethink if load test at that time is unsuccessful.

**Comment by EWashb** [comment]
TLDR- Decision made: YES to linky, however, more work needs to be done to write the script that converts all existing external links

**Comment by tjheffner** [comment]
we havent determined 'yes' to linky yet, only the steps forward to load test correctly. we think it is a 'yes' for the cms, but the results of the load test will determine if linky module is appropriate for us without detriment to downstream consumers (FE builds, jsonapi for AP)
